### PR TITLE
feat: implement version check

### PR DESCRIPTION
**Pull request by 10ko**
Implements version check.
Default url is `https://garden.io/version`.
Fails silently.

### CODE REVIEW

File: garden-service/src/cli/helpers.ts [hunk 1]
```diff
@@ -207,3 +215,26 @@ export function failOnInvalidOptions(argv, ctx) {
...
+
+    if (showMessage || !configObj.lastVersionCheck) {
+      if (res.data.status === "OUTDATED") {
+        printHeader(logger, res.data.message, "warning")
+      } else {
+        printHeader(logger, res.data.message, "thumbsup")
+      }
+      await config.set([globalConfigKeys.lastVersionCheck], { lastRun: new Date() })
+    }
+  } catch (err) { /* Fail silently */ }
```
Comment (edvald) [inline garden-service/src/cli/helpers.ts]: Maybe add a verbose log here instead?

File: garden-service/src/constants.ts [hunk 2]
```diff
@@ -38,3 +38,5 @@ export const SUPPORTED_PLATFORMS: SupportedPlatform[] = ["linux", "darwin", "win
 
 export const SEGMENT_DEV_API_KEY = "D3DUZ3lBSDO3krnuIO7eYDdtlDAjooKW"
 export const SEGMENT_PROD_API_KEY = "b6ovUD9A0YjQqT3ZWetWUbuZ9OmGxKMa"
+
+export const VERSION_CHECK_URL = "https://garden.io/version"
```
Comment (edvald) [inline garden-service/src/constants.ts]: Shouldn't this be `get.garden.io`?

File: garden-service/src/config-store.ts [hunk 3]
```diff
@@ -265,8 +270,14 @@ const AnalyticsGlobalConfigSchema = joi.object()
     firstRun: joi.boolean().optional(),
   }).meta({ internal: true })
 
+const VersionCheckGlobalConfigSchema = joi.object()
```
Comment (edvald) [inline garden-service/src/config-store.ts]: We generally keep the first char lowercase in variable names.

**Comment by 10ko** [comment]
Thanks for the comments @edvald.
Fixed as requested

**Pull request merged**